census.go

```diff
@@ -468,6 +417,89 @@ func (v *vocdoniHandler) censusFollowers(msg *apirest.APIdata, ctx *httprouter.H
 	return ctx.Send(data, http.StatusOK)
 }
 
+// censusCommunity creates a new census from a community. The census of the
+// community can be of type channel, NFT, or ERC20. If the community is a
+// channel, the census is created from the users who follow the channel, and
+// the process is async. If the community is an NFT or ERC20, the census is
+// created from the token holders of the token addresses in the community, using
+// the AirStack API. The process is sync and the census is created in the same
+// request. The census is created from the participants and the progress is
+// updated in the queue.
+func (v *vocdoniHandler) censusCommunity(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	// extract userFID from auth token
+	userFID, err := v.db.UserFromAuthToken(msg.AuthToken)
+	if err != nil {
+		return fmt.Errorf("cannot get user from auth token: %w", err)
+	}
+	req := struct {
+		CommunityID uint64 `json:"communityID"`
+	}{}
+	if err := json.Unmarshal(msg.Data, &req); err != nil {
+		return err
+	}
+	// get the community from the database
+	community, err := v.db.Community(req.CommunityID)
+	if err != nil {
+		return ctx.Send([]byte("error getting community"), http.StatusInternalServerError)
+	}
+	if community == nil {
+		return ctx.Send([]byte("community not found"), http.StatusNotFound)
+	}
+	// create a censusID for the queue and store into it
+	censusID, err := v.cli.NewCensus(api.CensusTypeWeighted)
+	if err != nil {
+		return err
+	}
+	// check the type to create it from the correct source (channel or airstak
+	// (nft/erc20)) and in the correct way (async or sync)
+	if community.Census.Type == mongo.TypeCommunityCensusChannel {
+		// if the census type is a channel, create the census from the users who
+		// follow the channel, the process is async so return add the censusID
+		// to the queue and return it to the client
+		data, err := v.censusWarpcastChannel(censusID, community.Census.Channel)
+		if err != nil {
+			log.Warnf("error creating census for the chanel: %s: %v", community.Census.Channel, err)
+			return ctx.Send([]byte("error creating channel census"), http.StatusInternalServerError)
+		}
+		return ctx.Send(data, http.StatusOK)
+	}
+	// if the census type is not a channel, the type is NFT or ERC20, so create
+	// the census sync and from the token holders from airstack
+	censusAddresses := []*CensusToken{}
+	for _, addr := range community.Census.Addresses {
+		censusAddresses = append(censusAddresses, &CensusToken{
+			Address:    addr.Address,
+			Blockchain: addr.Blockchain,
+		})
+	}
+	// check valid token
+	if err := v.checkTokens(censusAddresses); err != nil {
+		return err
+	}
+	// convert the census type to the correct type for the CreateCensus function
+	censusType := ERC20type
```

Comment:
Nit: a switch case works better here and it's more extensible to other censuses in the future